### ISSUE TITLE
Disabling the script causes Kodi to crash

### ISSUE THREAD

**Issue by paulnd** [sanghviharshit/script.kodi.lifx.ambilight#30]
* **I'm submitting a ...**
  - [x] bug report
  - [ ] feature request
  - [ ] feedback

* **If the current behavior is a bug, please provide the steps to reproduce**

Disable the script

  * **What is the current behavior?**

Kodi crashes

  * **What is the expected behavior?**

Kodi shouldn't crash

  * **(Important) Logs**
    * _(Follow https://github.com/sanghviharshit/script.kodi.lifx.ambilight/wiki/Logs for sharing link to logs)_
https://pastebin.com/BMN47gvY

* **What is the motivation / use case for changing the behavior?**

* **Please tell me about your environment:**
  - Kodi Version 17.6 (Windows)
  - Addon Version: 2.7.13
  - OS: Windows 10
  - Lifx Devices: in the log

* **Other information**
  * _(e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, etc.)_

**Comment by paulnd** [comment]
Hello, could this be looked at please? 

I hope that development of this script hasn't stopped. :worried:

**Comment by sanghviharshit** [comment]
Hey, can you please upload the logs again, it's not available anymore

**Comment by paulnd** [comment]
Hi, here's a new one from today:

https://pastebin.com/cr8sRKtS

**Comment by sanghviharshit** [comment]
Looking at the logs, I think what's happening is you are trying to disable the script while it's still initializing, leading to ```14:30:39.217 T:5676   ERROR: CPythonInvoker(1, C:\Users\Admin\AppData\Roaming\Kodi\addons\script.kodi.lifx.ambilight\service.py): script didn't stop in 5 seconds - let's kill it```
Unfortunately, the way Lifx device discovery works, it takes longer than usual for the script to respond to disable request and causes Kodi to terminate it.

Do you know if this still happens when the addon is fully initialized and then trying to disable it causes Kodi to crash?

**Comment by paulnd** [comment]
> Do you know if this still happens when the addon is fully initialized and then trying to disable it causes Kodi to crash?

Yes, it does. Here's a new log where about 6 minutes elapsed between starting the script and then disabling it, the error message is the same.

https://pastebin.com/i68FVu2M

**Comment by sanghviharshit** [comment]
I don't see any obvious issues in the logs. But I will look into improvements soon, but can't really promise anything :(
